packages/integration-objects/srv/integration-objects/simple-exchange-rate-type-service.js

```diff
@@ -0,0 +1,210 @@
+  srv.before('UPDATE', 'ExchangeRateTypeDescriptions', beforeUpdateForRateTypeDescriptions);
+  srv.before('DELETE', 'ExchangeRateTypeDescriptions', beforeActiveDeleteForExchangeRateTypeDescription);
+
+  srv.on('CREATE', 'ExchangeRateTypes', writeAuditLogForRateTypes);
+  srv.on('DELETE', 'ExchangeRateTypes', writeAuditLogForRateTypes);
+  srv.on('UPDATE', 'ExchangeRateTypes', writeAuditLogForRateTypes);
+
+  const { ExchangeRateTypes, ExchangeRateTypeDescriptions } = srv.entities;
+
+  function beforeRead(req) {
+    try {
+      checkAndAppendTenantIdFilterForReadEvent(req);
+    } catch (err) {
+      throw new ValidationError(err.message, err.code);
+    }
+  }
+
+  async function beforeCreate(req) {
+    try {
+      await validatePayload(req);
+    } catch (err) {
+      throw new ValidationError(err.message, err.code);
+    }
+  }
+
+  async function beforeUpdate(req) {
+    try {
+      await validatePayload(req);
+    } catch (err) {
+      throw new ValidationError(err.message, err.code);
+    }
+  }
+
+  async function validatePayload(req) {
+    validateTenantIdInPayload(req);
+    validateFieldsForExchangeRateTypes(req.data);
+    await checkUniquenessForPrimaryKeys(req);
+  }
+
+  function beforeDelete(req) {
+    try {
+      validateDelete(req);
+    } catch (err) {
+      throw new ValidationError(err.message, err.code);
+    }
+  }
+
+  async function beforeCreateForRateTypeDescriptions(req) {
+    try {
+      await validateDescPayload(req);
+    } catch (err) {
+      throw new ValidationError(err.message, err.code);
+    }
+  }
+
+  async function beforeUpdateForRateTypeDescriptions(req) {
+    try {
+      await validateDescPayload(req);
+    } catch (err) {
+      throw new ValidationError(err.message, err.code);
+    }
+  }
+
+  async function validateDescPayload(req) {
+    validateFieldsForExchangeRateTypeDescription(req.data, req.params);
+    await checkForValidityOfKey(req, ExchangeRateTypes);
+    await checkUniquenessForPrimaryKeysForExchangeRateTypeDescForUpdate(req);
+  }
+
+  async function checkUniquenessForPrimaryKeys(req) {
+    const record = req.data;
+    const affectedRows = await fetchExistingRateTypes(ExchangeRateTypes, record, req);
+    checkForDuplicateRateTypeObjects(affectedRows, req);
+  }
+
+  async function checkUniquenessForPrimaryKeysForExchangeRateTypeDescForUpdate(req) {
+    const record = req.data;
+    const affectedRows = await fetchExistingRateTypeDesc(ExchangeRateTypeDescriptions, record, req);
+    checkForDuplicateRateTypeObjects(affectedRows, req);
+  }
+
+  async function beforeActiveDeleteForExchangeRateTypeDescription(req) {
+    const record = req.data;
+    let id;
+    const affectedRows = await fetchExistingTexts(ExchangeRateTypeDescriptions, record, req);
+    if (affectedRows.length) {
+      id = affectedRows[0].ID_texts;
+    }
+    await validateIdInTenant(id, ExchangeRateTypes, req, affectedRows);
```

Comment:
See my comment below for `validateIdInTenant`:
```suggestion
    const affectedRows = await fetchExistingTexts(ExchangeRateTypeDescriptions, record, req);
    if (affectedRows.length) {
      await validateIdInTenant(affectedRows[0].ID_texts, ExchangeRateTypes, req, affectedRows);
    }
    
```